src/core/layout/qgslayoutexporter.cpp

```diff
@@ -1466,6 +1436,169 @@ bool QgsLayoutExporter::georeferenceOutputPrivate( const QString &file, QgsLayou
   return true;
 }
 
+QString nameForLayerWithItems( const QList< QGraphicsItem * > items, unsigned int layerId )
+{
+  if ( items.count() == 1 )
+  {
+    if ( QgsLayoutItem *layoutItem = dynamic_cast<QgsLayoutItem *>( items.at( 0 ) ) )
+      return layoutItem->displayName();
```

Comment:
I commented in your added tests on the less-than-ideal mixture of layer names using lesser than / greater than symbols vs. names not surrounded by those symbols. E.g. Page, Labels, scalebar, Map 1: lines, <Label>. IMO, it'd be worth fixing this if possible.